optima/optimization.py

```diff
@@ -96,18 +110,54 @@ def defaultobjectives(which='outcome', verbose=2):
     return objectives
 
 
+def defaultconstraints(project=None, progset=None, which='outcome', verbose=2):
+    """
+    Define default objectives for the optimization. Some objectives are shared
+    between outcome and money minimizations, while others are different. However,
+    outcome minimization is performed as part of money minimization, so it's useful
+    to keep all the keys for both. Still, ugly.
+    
+    Version: 2016feb03
+    """
+
+    printv('Defining default objectives...', 3, verbose=verbose)
+    
+    if type(progset)==Programset: pass
+    elif type(project)==Programset: progset = project
+    elif project is not None:
+        if progset is None: progset = 0
+        progset = project.progsets[progset]
+    else:
+        raise OptimaException('To define constraints, you must supply a program set as an input')
+
+    constraints = odict() # Dictionary of all constraints
+    constraints['name'] = odict() # Full name
+    constraints['min'] = odict() # Minimum budgets
+    constraints['max'] = odict() # Maximum budgets
+    for prog in progset.programs.values():
+        constraints['name'][prog.short] = prog.name
+        if prog.optimizable():
+            constraints['min'][prog.short] = 0.0
```

Comment:
i wonder whether this should eventually be a property of programs? like how limits are a property of parameters. although, maybe that wouldn't work generally since programs are not fixed like parameters...